transport/tchannel/config.go

```diff
@@ -32,11 +33,24 @@ import (
 
 const transportName = "tchannel"
 
+var defaultConnectionTimeout = 500 * time.Millisecond
+
 // TransportConfig configures a shared TChannel transport. This is shared
 // between all TChannel outbounds and inbounds of a Dispatcher.
 //
-// TransportConfig does not have any parameters at this time.
-type TransportConfig struct{}
+//  transports:
+//    tchannel:
+//      connection-timeout: 500ms
+type TransportConfig struct {
+	ConnectionTimeout time.Duration `config:"connection-timeout,interpolate"`
```

Comment:
We now have precedent for kabab-case with `round-robin` and `least-pending`. Yab templates also use kebab-case, but also support germancase.